posthog/cloud_utils.py

```diff
@@ -38,3 +44,44 @@ def is_cloud():
 def TEST_clear_cloud_cache(value: Optional[bool] = None):
     global is_cloud_cached
     is_cloud_cached = value
+
+
+def get_cached_instance_license() -> Optional["License"]:
+    """Returns the first valid license and caches the value for the lifetime of the instance, as it is not expected to change.
+    If there is no valid license, it returns None.
+    """
+    global instance_license_cached
+    global is_instance_licensed_cached
+
+    try:
+        from ee.models.license import License
+    except ProgrammingError:
+        # TRICKY - The license table may not exist if a migration is running
+        pass
+    except Exception as e:
+        capture_exception(e)
+        return None
+
+    if isinstance(instance_license_cached, License):
+        return instance_license_cached
+
+    if is_instance_licensed_cached is False:
+        # This is an unlicensed instance
+        return None
```

Comment:
It doesn't look like `is_instance_licensed_cached` is ever set to `False` outside of `TEST_clear_instance_license_cache` 🤔

Reply:
Whoops! Good catch, thanks.